src/db_state.rs

```diff
@@ -388,6 +395,8 @@ impl DbState {
                 next_tick: memtable_tick,
             });
         }
+        // update the persisted manifest last_l0_seq as the latest seq in the imm
+        state.core.last_l0_seq = cmp::max(state.core.last_l0_seq, imm_memtable.table().last_seq());
```

Comment:
This makes me a little nervous. Shouldn't `state.core.last_l0_seq` always be <= `imm_memtable.table().last_seq()`. If we ever hit a point where that's not true, we've got a problem.

Is this logic trying to handle the case where `last_seq()` [is 0 because no rows were added](https://github.com/slatedb/slatedb/pull/389/files#diff-4c311c545b79b549477a9f7d75b17af588efe93f7a4ffaca04fa3c2b3f9a5ae9R241) to the imm_memtable?

Reply:
yeah, i am also a bit worried about this, i think using an `Option` might be better than a zero value, let me have another try.

Reply:
replaced the `last_seq` as `Option` in 1ca8bce, but it feels a bit ugly to have a `Mutex` around `Option<u64>`, as `KvTable` is wrapped inside a `Arc` we have to use interior mutability to update its inner fields.

another possible way might be wrap the entire `KvTable` with a Mutex, but i dunno whether it's worth or not...

Reply:
Yea, these options all seem a bit icky. Can we keep it AtomicU64 and do:

```rust
        let memtable_last_seq = imm_memtable.table().last_seq();
        if memtable_last_seq > 0 {
            state.core.last_l0_seq = memtable_last_seq;
        }
```

Or maybe just do us `imm_memtable.table().is_empty()`?

Reply:
reverted to `AtomicU64` and still let `last_seq()` returns an `Option<u64>` for the caller do not require to understand the zero value of seq num in 9342e07, PTAL